facia-tool/app/tools/FaciaToolArchive.scala

```diff
@@ -0,0 +1,54 @@
+package tools
+
+import com.amazonaws.regions.{Regions, Region}
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsyncClient
+import com.amazonaws.services.dynamodbv2.model._
+import common.{ExecutionContexts, Logging}
+import conf.{Switches, Configuration}
+import org.joda.time.{DateTimeZone, DateTime}
+import play.api.libs.json._
+import scala.collection.JavaConverters._
+import awswrappers.dynamodb._
+
+import scala.util.{Failure, Success}
+
+case class ArchiveRequest(email: String, updateJson: JsValue, diff: JsValue)
+
+object FaciaToolArchive extends ExecutionContexts with Logging {
+  val TableName = "FaciaToolUpdateHistory"
+  private val dynamoClient: Option[AmazonDynamoDBAsyncClient] =
+    if (Configuration.environment.isProd) {
+      val c = new AmazonDynamoDBAsyncClient()
+      c.setRegion(Region.getRegion(Regions.EU_WEST_1))
+      Option(c)
+    } else None
+
+  def dayKey(date: DateTime) = date.toString("yyyy-MM-dd")
```

Comment:
@robertberry Yeah, I agree we could use the `collectionId` as the hash. Initially, I am just playing around with it and exploring what I need from it. The main question I wanted answered from this in this instance was who did what when, rather than who did what to this container.

In the future, I would imagine there would be more fields with more indexes; you can have multiple global indexes. I would also imagine in the ideal world, `facia-tool` wouldn't be the one doing the updates to the collections, and would just be feeding a stream, and the **thing** that hangs off that doing the updates would update dynamo also.

Were you getting throttling for writes or reads?